### PR TITLE
Minor fixes and improvements

### PR DESCRIPTION
**Pull request by mikatammi**

### CODE REVIEW

**Comment by nesteroff** [comment]
This is great. By the way, it looks like NXP has finally moved from Code Aurora to [GitHub](https://github.com/nxp-imx/). Do you think we should update [imx-atf](https://github.com/nxp-imx/imx-atf) and [imx-uboot](https://github.com/nxp-imx/uboot-imx) to download the sources from GitHub as well?

File: overlays/patches/0001-Fix-console-rootfs-for-imx8qm.patch [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+index 4588033..cd544f2 100644
 --- a/Makefile
 +++ b/Makefile
 @@ -37,6 +37,7 @@ FILES = \
-	etc/s6-linux-init/run-image/service/getty-tty3/run \
-	etc/s6-linux-init/run-image/service/getty-tty4/run \
-	etc/s6-linux-init/run-image/service/getty-ttyS0/run \
+ 	etc/s6-linux-init/run-image/service/getty-tty3/run \
+ 	etc/s6-linux-init/run-image/service/getty-tty4/run \
+ 	etc/s6-linux-init/run-image/service/getty-ttyS0/run \
```
Comment (unbel13ver) [inline overlays/patches/0001-Fix-console-rootfs-for-imx8qm.patch:8]: These spaces are added by git itself. I believe it is OK to merge, nevertheless git will warn about whitespaces. It is a known git behavior.

**Comment by unbel13ver** [comment]
Hey, good work and great example of "sterile" clean PR!
Would you mind if I squash `fetchGit` commits into one when merge?

**Review by unbel13ver — APPROVED** [review]

**Comment by vilvo** [comment]
- Please use `@unikie.com` over personal for work related commits in  `Author:`
- Please add the rationale in commit messages for the future readers/reviewers:
```
Please don't use builtins.fetchGit if you don't really need to. It happens during
evaluation time and not build time (makes it awfully slow), and uses the
systems git instead of nixpkgs git, and also makes nix flakes complain about
impurity (at least if you don't provide proper revision).

Also it's more reasonable to use fetchFromGitHub for the kernel, because it
downloads the snapshot .zip from GitHub, instead of downloading the whole
git commit history for the kernel.
```

**Review by vilvo — CHANGES_REQUESTED** [review]
Agreed on changes with the author.

**Comment by mikatammi** [comment]
> This is great. By the way, it looks like NXP has finally moved from Code Aurora to [GitHub](https://github.com/nxp-imx/). Do you think we should update [imx-atf](https://github.com/nxp-imx/imx-atf) and [imx-uboot](https://github.com/nxp-imx/uboot-imx) to download the sources from GitHub as well?

I'd say let's do this in another commit, as changing u-boot and atf versions might require some testing. This PR retains the previous functionality and commit versions, but just changes to use fetchFromGitHub/fetchgit instead of builtins.fetchGit

**Comment by mikatammi** [comment]
Had to change GPG key for signing, now it looks good and is ready for merging from my side

**Review by vilvo — APPROVED** [review]

**Pull request merged**